docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,921 @@
+
+The effect of the trend above is the blurring of the lines not only between different Cluster API components, but also
+between Cluster API, core Kubernetes and a few other broadly adopted tools like Helm or Flux (and to some extents,
+also with many others awesome tools in the ecosystem).
+
+This is why Cluster API status must become simpler to understand for users, and also more consistent not only across
+different CAPI resources, but with Kubernetes core and ideally with the entire ecosystem.
+
+### Goals
+
+- Review and standardize the usage of the concept of readiness across Cluster API resources.
+    - Drop or amend improper usage of readiness
+    - Make the concept of Machine readiness extensible, thus allowing providers or external system to inject their readiness checks.
+- Review and standardize the usage of the concept of availability across Cluster API resources.
+    - Make the concept of Cluster Availability extensible, thus allowing providers or external system to inject their availability checks.
+- Bubble up more information about both CP and worker Machines, ensuring consistent way across Cluster API resources.
+    - Standardize replica counters and bubble them up to the Cluster resource.
+    - Standardize control plane, MachineDeployment, Machine pool availability, and bubble them up to the Cluster resource.
+- Introduce missing signals about connectivity to workload clusters, thus enabling to mark all the conditions
+  depending on such connectivity being working with status Unknown after a certain amount of time.
+- Introduce a cleaner signal about Cluster API resources lifecycle transitions, e.g. scaling up or updating.
+- Ensure everything in status can be used as a signal informing monitoring tools/automation on top of Cluster 
+  about lifecycle transitions/state of the Cluster and the underlying components as well.
+
+### Non-Goals/Future Work
+
+- Resolving all the idiosyncrasies that exists in Cluster API, core Kubernetes, the rest of the ecosystem.
+  (Let’s stay focused in Cluster API and keep improving incrementally).
+
+## Proposal
+
+This proposal groups a set of changes to status fields in Cluster API resources.
+
+Some of those changes could be considered straight forward, e.g.
+
+- K8s API conventions suggest to deprecate and remove `phase` fields from status, Cluster API is going to align to this recommendation
+  (and improve Conditions to provide similar or even a better info as a replacement).
+- K8s resources do not have a concept similar to "terminal feature" existing in Cluster API resources, and users approaching
+  the project are struggling with this idea; in some cases also provider's implementers are struggling with it.
+  Accordingly, Cluster API resources are dropping `FailureReason` and `FailureMessage` fields (terminal failures should be surfaced using
+  conditions, like any other error/warning/message)
```

Comment:
Agreed on this one, the terminal failures today at least are often "solved" through MachineHealthCheck in some ways.

Reply:
Thanks for the feedback, resolving to keep the number of active threads under control 😅